neuraxle/steps/flow.py

```diff
@@ -326,29 +287,56 @@ class ChooseOneOrManyStepsOf(FeatureUnion):
         :class:`Optional`
     """
 
-    def __init__(self, steps, hyperparams=None):
+    def __init__(self, steps):
         FeatureUnion.__init__(self, steps)
-
-        if hyperparams is None:
-            self.set_hyperparams(HyperparameterSamples({}))
-        else:
-            self.set_hyperparams(hyperparams)
-
+        self.set_hyperparams(HyperparameterSamples({}))
         self._make_all_steps_optional()
 
     def _make_all_steps_optional(self):
         """
         Wrap all steps with :class:`Optional` wrapper.
         """
         step_names = list(self.keys())
-        for step_name in step_names:
+        for step_name in step_names[:-1]:
```

Comment:
are you sure -1 ? idk what it do at all, but it seems weird while reviewing.

Reply:
my unit tests wouldn't pass otherwise.